core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -171,7 +170,7 @@ object RequestChannel extends Logging {
       s"$forwardDescription$header -- ${loggableRequest.toString(details)}"
     }
 
-    def body[T <: AbstractRequest](implicit classTag: ClassTag[T], @nowarn("cat=unused") nn: NotNothing[T]): T = {
+    def body[T <: AbstractRequest](implicit classTag: ClassTag[T]): T = {
```

Comment:
In new Scala version, it will check `@nowarn` annotation, and make that build error. 
```
[2024-05-16T08:37:59.531Z] [Error] /home/jenkins/jenkins-agent/workspace/Kafka_kafka-pr_PR-15958/core/src/main/scala/kafka/network/RequestChannel.scala:174:69: @nowarn annotation does not suppress any warnings
[2024-05-16T08:37:59.531Z] one error found
```
In previous comments, we have two way to resolve this.
1. "-Wconf:cat=unused-nowarn:s" 
2. Remove the usage of NotNothing
we decide to use the second way to resolve this problem

Reply:
> It was there for a reason (to prevent cases where ClassTag is automatically inferred to Nothing)

That was a good protection, but we use the request once casting the `body`. Hence, the `nothing` type (i.e request.body`) will be noticed soon as we can't call any methods from it. 

> we decide to use the second way to resolve this problem

As @m1a2st mentioned, there is another approach for it. We can go back to use it if @ijuma prefer it